src/shared/debug/launchConfiguration.ts

```diff
@@ -84,3 +94,60 @@ class DefaultDebugConfigSource implements DebugConfigurationSource {
         await this.launch.update('configurations', value)
     }
 }
+
+function getExistingSamTemplateTargets(launchConfig: LaunchConfiguration): TemplateTargetProperties[] {
+    return _(launchConfig.getSamDebugConfigurations())
+        .map(samConfig => samConfig.invokeTarget)
+        .filter(isTemplateTargetProperties)
+        .value()
+}
+
+function getExistingSamCodeTargets(launchConfig: LaunchConfiguration): CodeTargetProperties[] {
+    return _(launchConfig.getSamDebugConfigurations())
+        .map(samConfig => samConfig.invokeTarget)
+        .filter(isCodeTargetProperties)
+        .value()
+}
+
+/**
+ * Returns a Set containing the samTemplateResources from the launch.json file that the launch config is scoped to.
+ * @param launchConfig Launch config to check
+ */
+export function getExistingSamTemplateResourcesForUri(launchConfig: LaunchConfiguration): Set<string> {
```

Comment:
nit: in similar fashion, consider something like getReferencedTemplatePaths  and  getReferencedCodeHandlers